src/libsync/owncloudpropagator_p.h

```diff
@@ -62,6 +63,52 @@ inline QByteArray getEtagFromReply(QNetworkReply *reply)
     return ret;
 }
 
+inline QPair<QByteArray, QByteArray> getExceptionFromReply(QNetworkReply *reply)
+{
+    Q_ASSERT(reply);
+    if (!reply) {
+        qDebug() << "Could not parse null reply";
+        return {};
+    }
+    const auto httpStatusCode = reply->attribute(QNetworkRequest::HttpStatusCodeAttribute).toInt();
+    // only for BadRequest and UnsupportedMediaType
+    if (httpStatusCode != 400 && httpStatusCode != 415) {
+        return {};
+    }
+
+    // we must not modify the buffer
```

Comment:
then why not making it clear by adding const to the declaration of `reply` ?
that would remove the warning and be guaranteed by the compiler (often better than me at guessing those things)

Reply:
@mgallien Can not do. See declaration of `QByteArray QIODevice::peek(qint64 maxlen)` is nonconst.